lib/stdlib/src/sofs.erl

```diff
@@ -345,12 +437,24 @@ relation(_) ->
     erlang:error(badarg).
 
 -doc """
-Creates a [relation](`m:sofs#relation`). [`relation(R, T)`](`relation/2`) is
-equivalent to [`from_term(R, T)`](`from_term/2`), if T is a
-[type](`m:sofs#type`) and the result is a relation.
+Creates a [relation](`m:sofs#relation`).
+
+[`relation(R, T)`](`relation/2`) is equivalent to
+[`from_term(R, T)`](`from_term/2`), if T is a [type](`m:sofs#type`)
+and the result is a relation.
 
 If `Type` is an integer N, then `[{atom, ..., atom}])`, where the tuple size is N,
 is used as type of the relation.
+
+## Examples
+
+```erlang
+1> sofs:relation([{1,red},{2,green},{3,blue}], [{index,color}]).
```

Comment:
Why not show the return here? The examples here show what you can have as input, but don't show what the output is. Is there a better function than sofs:type/1 that can be used to show what releation actually does?

Reply:
maybe to_external?

Reply:
Yes, I will use `sofs:to_external/1`. The internal representation of sofs sets is opaque, so I am not supposed them. I will rewrite the example to:

```
1> S1 = sofs:relation([{3,blue},{2,green},{3,blue},{1,red}], [{index,color}]).
2> sofs:to_external(S1).
[{1,red},{2,green},{3,blue}]
3> sofs:type(S1).
[{index,color}]
4> sofs:type(sofs:relation([{1,a},{1,b}], 2)).
[{atom,atom}]
5> sofs:type(sofs:relation([], 3)).
[{atom,atom,atom}]
```